dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/dataitem/ProgramItemStageElement.java

```diff
@@ -81,11 +91,33 @@ public Object getSql( ExprContext ctx, CommonExpressionVisitor visitor )
         assumeStageElementSyntax( ctx );
 
         String programStageId = ctx.uid0.getText();
+
         String dataElementId = ctx.uid1.getText();
 
-        String column = visitor.getStatementBuilder().getProgramIndicatorDataValueSelectSql(
-            programStageId, dataElementId, visitor.getReportingStartDate(), visitor.getReportingEndDate(),
-            visitor.getProgramIndicator() );
+        int stageOffset = visitor.getStageOffset();
+
+        String column;
+
+        if ( isNonDefaultStageOffset( stageOffset ) )
+        {
+            if ( isRepeatableStage( visitor.getProgramStageService(), programStageId ) )
+            {
+                column = visitor.getStatementBuilder().getProgramIndicatorEventColumnSql( programStageId,
+                    Integer.valueOf( stageOffset ).toString(),
+                    "\"" + dataElementId + "\"",
```

Comment:
Use `SqlUtils.quote` instead of manual quoting. I know dataElementId cannot have special characters, but this method will also escape double quotes.

Reply:
done